coding-guidelines/coding-guidelines.md

```diff
@@ -0,0 +1,273 @@
+  "key3": "value3",
+  "key4": "value4"
+};
+```
+
+
+## Indentation and Line Breaks
+Rules:
+- Line breaks are only done if needed.
+- Strings are split at the latest possible position.
+  - At the latest whitespace character.
+  - The second line is indented with 2 spaces.
+  ```JS
+    var longString = "A Very long string which is longer then one line, which is 80 characters " +
+      "in this case. This demonstrates how the second line is indented";
+  ```
+- Object literals are kept on one line if possible.
+  - If the line gets to long put each attribute on a single line.
+  - Multi level object literals must be split in to multiple lines.
+    - Each level in the object must have 2 spaces as indentation.
+
+Examples:
+```JS
+test(
+  "A Very long string which is longer then one line, which is 80 characters " +
+    "in this case. This demonstrates how the second line is indented",
+  "Another string parameter.",
+  {and: "a Literal Object", as: "a parameter", type: "short"},
+  {
+    yet: "another long parameter Object",
+    which: "needs to be split in multiple",
+    lines: 3
+  }
+);
+
+test({
+  a: {
+    b: 10
+  },
+  c: 20,
+  d: 30,
+  e: 40,
+  f: 50,
+  g: 60,
+  h: {
+    i: 70
+  }
+});
+```
+
+## Functions
+Rules:
+  - Function blocks must have a space before the definition.
+  - Function must have a space before the opening parenthesis.
+  - Named functions must not have a space before the opening function brackets.
+  - Functions can be on a single line.
+    - If they are there must be a whitespace inside the curly brackets.
+  - Anonymous functions need to have a space before the opening function bracket.
+  - Arrow functions
+    - can be on a single line.
+    - must omit the parameter brackets if only one parameter is present.
+    - must omit curly brackets if only one statement is given.
+    - must have whitespace around the parameter definition
+
+Examples:
+```JS
+function test() {
+  return;
+}
+
+function test() { var a = 10; return a; }
+
+var test = function () { var b = 10; return b; }
+
+var es6 = parameter => parameter();
+
+var es6 = parameter => {
+  parameter();
+  return false;
+}
+
+var es6 = (parameter, index) => parameter(index);
+
+var es6 = (parameter, index) => {
+  parameter(index);
+  return false;
+}
+```
+
+## Control Statements
+Rules:
+  - Space after the keyword
+  - Space before opening parentheses
+  - Strict equal (`===` or `!==`)
+    - exception to this rule is null checking
+
+Examples:
+```JS
+if (a === b) {
+  a();
+}
+
+if (true !== false) {
+  // Statements
+}
+
+if (a == null) {
+  // Statements
+} else {
+  // Statements
+}
+
+if (true !== false) {
+  // Statements
+} else if(false === true) {
+  // Statements
+} else {
+  // Statements
+}
+
+switch (a) {
+  case 1:
+    break;
+  case 2:
+    break;
+  default:
+    break;
+}
+
+for (var a = 10; a !== 0; a--) {
+  // Statements
+}
+```
+
+## JSDoc
+JSDoc block should not be needed if they are needed to these rules apply:
```

Comment:
Rewrite maybe to:
`JSDoc blocks should be avoided in favor to code descriptiveness. If the are necessary these rules apply:`